compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -924,6 +939,25 @@ static LogicalResult setWarpReductionConfig(func::FuncOp entryPoint,
   if ((groupSize / subgroupSize) > subgroupSize)
     return failure();
 
+  // With just one subgroup per workgroup, make each subgroup do more work and
+  // process a few reductions (rows) along the last parallel dimension.
+  //
+  // TODO: This is enabled for matvec on ROCm for now. We should
+  // validate this strategy and extend to more linalg generics and to CUDA.
+  if (isRocmTarget(entryPoint) && groupSize == subgroupSize &&
+      isMatvecLike(op) && llvm::none_of(bounds, ShapedType::isDynamic)) {
+    int64_t lastParallelBound = bounds[parallelDims.back()];
```

Comment:
Not saying we need to do this in this PR, but couldn't we also do this with all parallel dimensions as opposed to jsut the last one?

Reply:
Same as above, see https://github.com/openxla/iree/pull/15941#discussion_r1427567016.